src/deploy/001_authenticator.ts

```diff
@@ -0,0 +1,25 @@
+import { DeployFunction } from "hardhat-deploy/types";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { salt, logResult, contractNames } from "../ts/deploy";
+
+const deployAuthenticator: DeployFunction = async function (
+  hre: HardhatRuntimeEnvironment,
+) {
+  const { deployments, getNamedAccounts, network } = hre;
+  const { deployer, owner } = await getNamedAccounts();
+  const { deploy, log } = deployments;
+
+  const { authenticator } = contractNames;
+
+  const deployResult = await deploy(authenticator, {
+    from: deployer,
+    gasLimit: 2000000,
+    args: [owner],
```

Comment:
Just curious, does this mean that right now the account that owns this deployment is `0x1111111...1111`?  Also, if we change the owner account, then we would deploy to a different address correct?

This means that if we want all of our contract addresses to be deployed to the same address, then we need to use the same owner account for all networks (but they don't need to have the same nonce)?

Reply:
You are correct.
The owner in any network that is not `hardhat` is currently `0x0000000...0001`.
The owner address is part of the EVM contract deployment code, so it's used to generate the contract address. This means we should use the same address on each network, but also that the nonce doesn't matter.
We could have a dedicated EOA whose only role is that of transferring ownership to the actual owner on each network. We could add this as a final step in the migration script and put the private key in an env variable. (I'd do this in case in another PR.)